src/ItemMeasurer.js

```diff
@@ -46,44 +45,163 @@ type ItemMeasurerProps = {|
   width: number,
 |};
 
+const scrollableContainerStyles = {
+  display: 'inline',
+  width: '0px',
+  height: '0px',
+  zIndex: '-1',
+  overflow: 'hidden',
+  margin: '0px',
+  padding: '0px',
+};
+
+const scrollableWrapperStyle = {
+  position: 'absolute',
+  flex: '0 0 auto',
+  overflow: 'hidden',
+  visibility: 'hidden',
+  zIndex: '-1',
+  width: '100%',
+  height: '100%',
+  left: '0px',
+  top: '0px',
+};
+
+const expandShrinkContainerStyles = {
+  flex: '0 0 auto',
+  overflow: 'hidden',
+  zIndex: '-1',
+  visibility: 'hidden',
+  left: '-9px',
+  bottom: '-8px',
+  right: '-8px',
+  top: '-9px',
+};
+
+const expandShrinkStyles = {
+  position: 'absolute',
+  flex: '0 0 auto',
+  visibility: 'hidden',
+  overflow: 'scroll',
+  zIndex: '-1',
+  width: '100%',
+  height: '100%',
+};
+
+const shrinkChildStyle = {
+  position: 'absolute',
+  height: '200%',
+  width: '200%',
+};
+
 export default class ItemMeasurer extends Component<ItemMeasurerProps, void> {
   _node: HTMLElement | null = null;
   _resizeObserver: ResizeObserver | null = null;
+  _resizeSensorExpand = React.createRef();
+  _resizeSensorShrink = React.createRef();
 
   componentDidMount() {
     const node = ((findDOMNode(this): any): HTMLElement);
     this._node = node;
-
     // Force sync measure for the initial mount.
     // This is necessary to support the DynamicSizeList layout logic.
     this._measureItem(true);
-
-    this._resizeObserver = new MutationObserver(this._onResize);
-    this._resizeObserver.observe(node, {
-      childList: true,
-      characterData: true,
-      subtree: true,
-    });
+    if (this.props.size) {
+      // Don't wait for positioning scrollbars when we have size
+      // This is needed triggering an event for remounting a post
+      this.positionScrollBars();
+    }
   }
 
   componentDidUpdate(prevProps) {
     if (prevProps.width !== this.props.width) {
       this._onResize();
     }
+
+    if (
+      (prevProps.size === 0 && this.props.size !== 0) ||
+      prevProps.size !== this.props.size
+    ) {
+      this.positionScrollBars();
+    }
   }
 
+  positionScrollBars = (height = this.props.size, width = this.props.width) => {
+    //we are position these hiiden div scroll bars to the end so they can emit
```

Comment:
Links here summarise on how we are triggering recalc based on scroll event when height changes